library/common/buffer/utility.cc

```diff
@@ -11,8 +11,6 @@ namespace Utility {
 Buffer::InstancePtr toInternalData(envoy_data data) {
   // This fragment only needs to live until done is called.
   // Therefore, it is sufficient to allocate on the heap, and delete in the done method.
-  // TODO: this method leaks the implementation of Buffer::BridgeFragment and could be improved to
```

Comment:
This is done by having the private constructor, and private destructor. And only allowing callers to allocate via the static method, and deallocate via done.